### PR TITLE
[release-4.16] NO-JIRA: Fixes for Keycloack IDP test and unit tests

### PR DESCRIPTION
**Pull request by liouk**
This is a manual cherry-pick of the following PRs:
- #679
- #680

Both are needed at the same time as both fix required tests (#679 fixes `e2e-operator`, #680 fixes `unit`).

### CODE REVIEW

File: test/library/idpdeployment.go [hunk 1]
```diff
@@ -71,19 +81,75 @@ func deployPod(
 		metav1.CreateOptions{},
 	)
 
+	saName := name
```
Comment (ibihim) [inline test/library/idpdeployment.go:84]: **nit:** if you want to create a copy of the name for the deployment and use it deliberately as the name of the ServiceAccount, I would do it above the creation of the ServiceAccount and use it in there:

```go
	saName := name
	_, err = clients.CoreV1().ServiceAccounts(namespace).Create(
		testContext,
		&corev1.ServiceAccount{
			ObjectMeta: metav1.ObjectMeta{
				Name: saName,
			},
		},
		metav1.CreateOptions{},
	)
```

File: test/library/idpdeployment.go [hunk 2]
```diff
@@ -71,19 +81,75 @@ func deployPod(
...
 	require.NoError(t, err)
 
 	_, err = clients.CoreV1().Services(namespace).Create(testContext, svcTemplate(httpPort, httpsPort), metav1.CreateOptions{})
 	require.NoError(t, err)
 
+	timeLimitedCtx, cancel := context.WithTimeout(testContext, 5*time.Minute)
+	defer cancel()
+	_, err = watchtools.UntilWithSync(timeLimitedCtx,
+		cache.NewListWatchFromClient(
+			clients.AppsV1().RESTClient(), "deployments", namespace, fields.OneTermEqualSelector("metadata.name", deployment.Name)),
```
Comment (ibihim) [inline test/library/idpdeployment.go:143]: Cool.

File: pkg/libs/endpointaccessible/endpoint_accessible_controller_test.go [hunk 3]
```diff
@@ -34,7 +34,7 @@ func Test_endpointAccessibleController_sync(t *testing.T) {
 		{
 			name: "non working endpoints",
 			endpointListFn: func() ([]string, error) {
-				return []string{"https://google.com", "https://nonexistenturl.com"}, nil
+				return []string{"https://httpbin.org/status/500"}, nil
```
[the remark this answers is not in the document]
Comment (liouk) [reply pkg/libs/endpointaccessible/endpoint_accessible_controller_test.go:37]: It was, anyway -- some domain sales service has picked up `https://nonexistenturl.com` and now it's working, when we expected it wouldn't :slightly_smiling_face:

File: test/library/idpdeployment.go (hunk 1 above)
Comment (liouk) [reply test/library/idpdeployment.go:84]: Good point, we missed this in the original PR -- but for the sake of consistency with the original (#679) let's not change it in the backport, WDYT?

File: pkg/libs/endpointaccessible/endpoint_accessible_controller_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (vrutkovs) [reply pkg/libs/endpointaccessible/endpoint_accessible_controller_test.go:37]: We rely on external service to return an error - HTTP 500 if its successful or 403/DNS error if its broken

**Pull request merged**